versionedController/v1/configuration/configuration.go

```diff
@@ -35,9 +35,9 @@ func New() *Controller {
 	}
 }
 
-//Put updates the password of the concerned user
+// Put updates the password of the concerned user
+// nolint: cyclop
```

Comment:
What does this mean?

Reply:
I'm telling golangci-lint to not run/report the cyclomatic complexity for this function as it exceeds the upper limit(it's 15) I had initially thought for a function(10). If I remove this comment the build will fail till this is rectified.

Reply:
(Q) Is it a nice idea to put some statements in a util function so that the skip linting rule be removed?
(A) Kind of, but that's a good short term solution to keep the CI happy, the cognitive load to understand the method won't reduce if we do that.

(Q) Do I feel strongly about cyclomatic complexity?
(A) Sort of, because testing will become difficult as there are too many code-paths for different cases, so the Put method test method will get longer and basically harder to read. Sometime in future I'd like to move the AuthZ logic out in a middleware so that core-logic & access-logic are separated out. IMO, it's just a metric we should consider while writing our logic and re-think if we can simplify things, if a single+simple business logic demands a switch with 15 cases, I'd ignore this metric.

## Background: GolangCI-Lint

It a [tool lifecycle manager & aggregator](https://golangci-lint.run/), i.e. we can configure each tools's properties in a single config file(TOML, YAML, etc) and it'd run the `n` tools which are enabled with their respective configurations.

## Background: Cyclomatic complexity

It's a good metric to evaluate the code complexity in terms of branching/conditional/loop/logical statements